cmd/osde2e/test/cmd.go

```diff
@@ -141,7 +141,8 @@ func init() {
 	viper.BindPFlag(config.Tests.SkipClusterHealthChecks, Cmd.PersistentFlags().Lookup("skip-health-check"))
 	viper.BindPFlag(config.Tests.GinkgoFocus, Cmd.PersistentFlags().Lookup("focus-tests"))
 	viper.BindPFlag(config.Tests.GinkgoSkip, Cmd.PersistentFlags().Lookup("skip-tests"))
-	viper.BindPFlag(config.MustGather, Cmd.PersistentFlags().Lookup("must-gather"))
+	// Cmd.PersistentFlags() fails to retrieve value of must-gather flag, the following line should be kept to Cmd.Flags() for accurate flag value
```

Comment:
@ritmun I Just pulled the latest osde2e code from `main` branch and everything seems to be working as expected. 

```shell
$ echo "Not supplying must gather and use what osde2e sets as a default"
Not supplying must gather and use what osde2e sets as a default

$ ./out/osde2e test --custom-config custom-rosa-hypershift-config.yml --configs hypershift
2023/06/01 08:33:36 configs.go:24: Will load config hypershift
2023/06/01 08:33:36 load.go:56: Custom YAML config provided, loading from custom-rosa-hypershift-config.yml
2023/06/01 08:33:36 cmd.go:166: Must gather: true <<<<
2023/06/01 08:33:36 cmd.go:167: Debugging, forced exit

$ echo "Supplying must gather as command line argument"
Supplying must gather as command line argument

$ ./out/osde2e test --custom-config custom-rosa-hypershift-config.yml --configs hypershift --must-gather=true
2023/06/01 08:33:54 configs.go:24: Will load config hypershift
2023/06/01 08:33:54 load.go:56: Custom YAML config provided, loading from custom-rosa-hypershift-config.yml
2023/06/01 08:33:54 cmd.go:166: Must gather: true <<<<
2023/06/01 08:33:54 cmd.go:167: Debugging, forced exit

$ echo "Supplying must gather as command line argument"
Supplying must gather as command line argument

$ ./out/osde2e test --custom-config custom-rosa-hypershift-config.yml --configs hypershift --must-gather=false
2023/06/01 08:34:03 configs.go:24: Will load config hypershift
2023/06/01 08:34:03 load.go:56: Custom YAML config provided, loading from custom-rosa-hypershift-config.yml
2023/06/01 08:34:03 cmd.go:166: Must gather: false <<<<
2023/06/01 08:34:03 cmd.go:167: Debugging, forced exit
```

```shell
$ ./out/osde2e test --custom-config custom-rosa-hypershift-config.yml --configs hypershift --must-gather false
2023/06/01 08:34:16 configs.go:24: Will load config hypershift
2023/06/01 08:34:16 load.go:56: Custom YAML config provided, loading from custom-rosa-hypershift-config.yml
2023/06/01 08:34:16 cmd.go:166: Must gather: true <<<<
2023/06/01 08:34:16 cmd.go:167: Debugging, forced exit
```

I do not get the value I provide if I do not include an `=` when defining the flag as seen above.

How are you setting the option from command line?

Reply:
I did print the value in my above examples. I just injected some debug code right after everything is loaded to quickly test. It simulates the same line `if viper.GetBool(config.MustGather)`.

```
2023/06/01 08:34:16 cmd.go:166: Must gather: true <<<<
```

You can see them with the `<<<<` next to the output.

I do get the default value when I try providing the value to the option without using `=`.

```
$ ./out/osde2e test --custom-config custom-rosa-hypershift-config.yml --configs hypershift --must-gather false
2023/06/01 08:34:16 configs.go:24: Will load config hypershift
2023/06/01 08:34:16 load.go:56: Custom YAML config provided, loading from custom-rosa-hypershift-config.yml
2023/06/01 08:34:16 cmd.go:166: Must gather: true <<<<
2023/06/01 08:34:16 cmd.go:167: Debugging, forced exit
```

From the readme doc, we document if using the flag, you provide value as `--must-gather=false`. Have you tried providing it like this? In my examples above, that does reflect the correct value.

This also makes me think, what if we rename this flag to be `--skip-must-gather`. This way when its not provided, it collects must gather artifacts by default. If a user would like to skip it, they simply just supply `--skip-must-gather`. Similar to how we have updated the `--skip-destroy-cluster` [flag](https://github.com/openshift/osde2e/pull/1642). What do you think?

Reply:
Sounds good to me 👍 

I just checked our prow jobs and we do set this for 3 jobs (1 periodic and 2 PR checks):

- [ROSA PR Check](https://github.com/openshift/release/blob/cb28205eb9de85a392b1608e74d3780b5702b0b3/ci-operator/config/openshift/osde2e/openshift-osde2e-main.yaml#L41)
- [ROSA HCP PR Check](https://github.com/openshift/release/blob/cb28205eb9de85a392b1608e74d3780b5702b0b3/ci-operator/config/openshift/osde2e/openshift-osde2e-main.yaml#L67)
- [ROSA HCP Periodic](https://github.com/openshift/release/blob/cb28205eb9de85a392b1608e74d3780b5702b0b3/ci-operator/config/openshift/osde2e/openshift-osde2e-main.yaml#L303)

We will want to open a secondary PR to address the environment variable rename.